### PR TITLE
changed to address issues 13-18

### LINKED ISSUE #13: Implement mutant strains
Just a placeholder for now. Francesco wanted to have this option, but he hasn't added any code to handle different strains.

_Originally posted by @fkang-pu in https://github.com/Princeton-LSI-ResearchComputing/neuron-simulator/pull/12#discussion_r835998064_

### LINKED ISSUE #14: The x-axis of the plot must be in seconds
### Feature Inspiration

Currently, the x-axis only displays time points. Since it is possible to change the the time scale of the plot (via `t_max`), showing the seconds on the x-axis provides the users with more information in more concrete terms than the abstract time points now displayed.

### Feature Description

* [ ] Multiplying the `time point` by `delta t` will convert the time points scale into seconds.
    ```
    seconds = time_point * (t_max / num_time_points)
    ```
* [ ] Change parameter text "Maximum time (t_max)" -> "Maximum time (s)"

### Comment

### LINKED ISSUE #15: Set the opacity of plot lines based on the confidence values
### Feature Inspiration

Plots used by the lab and likely in the publication will displaying neurons with low confidence as increasingly translucent as an intuitive way increase the information density of the plots. 

### Feature Description

* [ ] Each output neuron has a confidence value associated with it. Set the line opacity in plotly based on the confidence (0 = completely translucent, 1 = fully opaque)
* [ ] add legend text at the top of the plot to explain. Preferably, include a scale using white (0.0) -> gray (0.5) -> black (1.0) (dynamic would be nice, but static image is fine)

### Comment

### LINKED ISSUE #16: Update the top menu
### Feature Inspiration

Make it simple for users to find help, download software, and link to the related neuron-graph project

### Feature Description

* [ ] Help: Remove dropdown arrow, make link to GitHub Readme
* [ ] Add "Neuron Graph" as item linked to https://neuron-graph.princeton.edu/
* [ ] Link to GitHub: Change text to "Worm Functional Connectivity Python Software Library"

### Comment

### LINKED ISSUE #17: Ignore "Top N" value when "Responding Neurons" are selected
### Feature Inspiration

If a user has gone to the trouble of selecting a specific set of neurons, return those and ignore the "top n" setting. If a user does select a "top n" value now, their responding neuron selections are cleared, but they may not expect that and get annoyed that their carefully selected values are wiped out. 

### Feature Description

If a user selects a set of responding neurons in addition to a value for "top n", show a warning next to the "Top N Responses" parameter: "The value for Top N Responses is ignored when specific response neurons are selected".

Both values should be passed to the underlying library, which will be responsible for implementing the desired behavior (see https://github.com/leiferlab/worm-functional-connectivity/issues/6).

### Comment

### LINKED ISSUE #18: Change the default stimulus type to "realistic"
### Feature Inspiration

We anticipate that most users will be interested the "realistic" stimulus type.

### Feature Description

* [ ] Set the default value to "realistic"
* [ ] Change the parameter text from "Type of standard stimulus" to the preferred "Type of standard activity"

### Comments

### PR DESCRIPTION
**Pull request by fkang-pu**
## Summary Change Description

Implemented changes to address issues or proposals described in Issues 13 to 18.

Resovles

- Resolves #13
- Resolves #14
- Resolves #15
- Resolves #16
- Resolves #17
- Resolves #18

## Code Review Notes

- notes for issue #13
 Created a place holder file for mutant "unc-31". To test JavaScript code for changing neuron options when a different strain_type is selected, neuron choices for "unc-31" were truncated to 5 items on purpose. Notes that the web form display would change as expected, but the plots for "unc-31" might display different neurons as the pickled atlas was a place holder file (renamed after copying the file for wild type).

- notes for issue #15

  item 1: 
  plotly opacity parameter is equivalent to matplotlib alpha parameter.

  item 2: 
  the change I made was not exactly as proposed.
  After testing with a range of opacity values (0 to 1), I feel it's better to list confidence value for each scatter plot (added to legend and hover text now), instead of showing the range from 0 to 1 by image. Because if confidence values are not significantly different, it's a bit hard to get clear ideas of confidence values based on line opacity (especially each line has its own color).

- notes for issue issues #17:
 warning message for top N is already included in Worm Functional Connectively package. User selected resp_neu_ids are now kept on web form while the warning message for top N is displayed on top of plot section.

- added code for getting n colors based on total responding neurons.  Seems the "rainbow" corlor scale gives more distinct color ranges than other ployly built-in corlor templates. 

- Made changes to Django form. Now the `stim_neu_id` is not selected as default, since nueron choices are set dynamically based on strain_type. A message would be displayed if a user clicked "Plot" button without selecting a `stim_neu_id`.

### CODE REVIEW

**Review by lparsons — COMMENTED** [review]
Generally, looks great and gets this very close to done. I did have a couple of quick comments while I review the rest of things:
* There is a missing dependency (`pandas`): https://github.com/Princeton-LSI-ResearchComputing/neuron-simulator/runs/6263963533?check_suite_focus=true#step:6:60
* It would be preferable if there were individual PRs for each issue, since it would make the reviews much simpler. It would also make testing more straightforward.

File: neuronsimulator/forms.py [hunk 1]
```diff
@@ -14,53 +14,30 @@ class ParamForm(forms.Form):
 
     form_opt_field_dict = wfc2plot().get_form_opt_field_dict()
 
-    neuron_choices = [
-        (neu_id, neu_id)
-        for neu_id in Neuron.objects.all().values_list("name", flat=True)
-    ]
+    strains = wfc.strains
```
Comment (lparsons) [inline neuronsimulator/forms.py:17]: I like getting the strains from the underlying library, but does the library also map these strains to atlas files? If not, perhaps we want to include that mapping here? There is no point is listing a strain for which we have no atlas (data).

**Review by lparsons — CHANGES_REQUESTED** [review]
This looks very good to me

The one blocking issue I see is that `pandas` is missing from the `requirements.txt`. 

I like the call to display the confidence values instead of using opacity since I agree it would be very difficult to tell given the use of multiple colors.

**Comment by fkang-pu** [comment]
> The one blocking issue I see is that `pandas` is missing from the `requirements.txt`.
> 
I  added pandas to requirements/common.txt. Also upgraded plotly to the latest version (5.7.0). Fixed test.py due to changes made to the form and utils.py.

**Review by lparsons — APPROVED** [review]
lgtm

**Pull request merged**